nuclio_sdk/logger.py

```diff
@@ -82,25 +83,33 @@ def set_handler(self, handler_name, file, formatter):
         self._handlers[handler_name] = stream_handler
 
     def debug(self, message, *args):
-        self._logger.debug(message, *args)
+        self._update_bound_vars_and_log(logging.DEBUG, message, *args)
 
     def info(self, message, *args):
-        self._logger.info(message, *args)
+        self._update_bound_vars_and_log(logging.INFO, message, *args)
 
     def warn(self, message, *args):
-        self._logger.warning(message, *args)
+        self._update_bound_vars_and_log(logging.WARNING, message, *args)
 
     def error(self, message, *args):
-        self._logger.error(message, *args)
+        self._update_bound_vars_and_log(logging.ERROR, message, *args)
 
     def debug_with(self, message, *args, **kw_args):
-        self._logger.debug(message, *args, extra={'with': kw_args})
+        self._update_bound_vars_and_log(logging.DEBUG, message, *args, **kw_args)
 
     def info_with(self, message, *args, **kw_args):
-        self._logger.info(message, *args, extra={'with': kw_args})
+        self._update_bound_vars_and_log(logging.INFO, message, *args, **kw_args)
 
     def warn_with(self, message, *args, **kw_args):
-        self._logger.warning(message, *args, extra={'with': kw_args})
+        self._update_bound_vars_and_log(logging.WARNING, message, *args, **kw_args)
 
     def error_with(self, message, *args, **kw_args):
-        self._logger.error(message, *args, extra={'with': kw_args})
+        self._update_bound_vars_and_log(logging.ERROR, message, *args, **kw_args)
+
+    def bind(self, **kw_args):
+        self._bound_variables.update(kw_args)
+
+    def _update_bound_vars_and_log(self, level, message, *args, **kw_args):
+        kw_args.update(self._bound_variables)
+
+        self._logger._log(level, message, args, extra={'with': kw_args})
```

Comment:
that will turn every usage of `info` and `info_with` to `info_with`
consider checking if kwargs is not empty and then provide it with  `extra`
something like `extra = {'with': kw_args } if kw_args else None`